sklearn/preprocessing/_target_encoder.py

```diff
@@ -0,0 +1,270 @@
+    scheme mixes the global target mean with the target mean conditioned on
+    the value of the category. [MIC]_
+
+    Read more in the :ref:`User Guide <target_encoder>`.
+
+    .. note::
+        :class:`TargetEncoder` uses a cross validation scheme in
+        :meth:`fit_transform` to prevent leaking the target during training. In
+        :meth:`fit_transform`, the data is split according to the `cv` parameter.
+        Categorical encodings are learned from split and used to encode the other split.
+        Afterwards, a final categorical encoding is learned from all the data, which
+        is then used to encode data during :meth:`transform`. This means that
+        `fit(X, y).transform(X)` does not equal `fit_transform(X, y)`.
+
+    .. versionadded:: 1.3
+
+    Parameters
+    ----------
+    categories : "auto" or a list of array-like, default="auto"
+        Categories (unique values) per feature:
+
+        - `"auto"` : Determine categories automatically from the training data.
+        - list : `categories[i]` holds the categories expected in the i-th column. The
+          passed categories should not mix strings and numeric values within a single
+          feature, and should be sorted in case of numeric values.
+
+        The used categories is stored in the `categories_` fitted attribute.
+
+    target_type : {"auto", "continuous", "binary"}, default="auto"
+        Type of target
+
+        - `"auto"` : Type of target is inferred with
+          :func:`~sklearn.utils.multiclass.type_of_target`
+        - `"continuous"` : Continuous target
+        - `"binary"` : Binary target
+
+    smooth : "auto" or float, default="auto"
+        The amount of mixing of the categorical encoding with the global target mean. A
+        larger `smooth` value will put more weight on the global target mean.
+        If `"auto"`, then `smooth` is estimated using an empirical bayes estimate.
+
+    cv : int, cross-validation generator or an iterable, default=None
+        Determines the cross-validation splitting strategy used in
+        :meth:`fit_transform`. Possible inputs for cv are:
+
+        - `None`, to use the default 5-fold cross validation,
+        - integer, to specify the number of folds in a `(Stratified)KFold`,
+        - :term:`CV splitter`,
+        - An iterable yielding (train, test) splits as arrays of indices.
+
+        Refer :ref:`User Guide <cross_validation>` for the various cross-validation
+        strategies that can be used here.
+
+    Attributes
+    ----------
+    encodings_ : list of shape (n_features,) of ndarray
+        For feature `i`, `encodings_[i]` is the encoding matching the
+        categories listed in `categories_[i]`.
+
+    categories_ : list of shape (n_features,) of ndarray
+        The categories of each feature determined during fitting
+        (in order of the features in `X` and corresponding with the output
+        of :meth:`transform`).
+
+    target_type_ : str
```

Comment:
Somehow I think that we should add some more documentation highlighting that target type inference can fail.

Reply:
I added a note to the `target_type` parameter about how the automatic target type inference may not be desired.